core/arch/arm/kernel/thread_spmc.c

```diff
@@ -350,46 +345,102 @@ static bool is_nil_uuid(uint32_t w0, uint32_t w1, uint32_t w2, uint32_t w3)
 
 static bool is_optee_os_uuid(uint32_t w0, uint32_t w1, uint32_t w2, uint32_t w3)
 {
-	return w0 == OPTEE_MSG_OS_OPTEE_UUID_0 &&
-	       w1 == OPTEE_MSG_OS_OPTEE_UUID_1 &&
-	       w2 == OPTEE_MSG_OS_OPTEE_UUID_2 &&
-	       w3 == OPTEE_MSG_OS_OPTEE_UUID_3;
+	return w0 == TEE_U32_FROM_BIG_ENDIAN(OPTEE_MSG_OS_OPTEE_UUID_0) &&
```

Comment:
This is bound to cause a change in the ABI towards normal world.

Reply:
We need this to be RFC 4122 compliant. It has been changed in TF-A as well.
https://git.trustedfirmware.org/TF-A/trusted-firmware-a.git/commit/?id=0d4120d87e9cb2f48374909068ad6b9325e25c9c 
Normally Sudeeps Linux driver should be aligned with this.

Reply:
I think this should be coordinated with https://github.com/OP-TEE/optee_os/pull/4509 and in particular the commit "core: spmc: FF-A ABI updates" in order to keep track of ABI changes.

Reply:
Rebased